syntax = "proto3";

package nighthawk.client;

import "api/client/options.proto";
import "api/client/output.proto";

message ExecutionRequest {
  enum CommandType {
    UNSET = 0;
    START = 1;
    UPDATE = 2;
  }
  CommandType command_type = 1;
  oneof command_specific_options {
    CommandLineOptions options = 2;
  }
}

message ExecutionResponse {
  Output output = 1;
}

service NighthawkService {

  /*
  Accepts a ExecutionRequest, which can be used to initiate, update, or cancel a test.
  Returns a stream of ExecutionResponse objects which will contain results for all successfull
  NH runs.
  */
  rpc sendCommand(stream ExecutionRequest) returns (stream ExecutionResponse) {
  }
}
